base/statistics.jl

```diff
@@ -388,18 +389,26 @@ cov(X::AbstractMatrix, Y::AbstractMatrix) = cov(X, Y, 1, true)
 
 ##### correlation #####
 
+"""
+    clampcor(x)
+
+Clamp a real correlation to between -1 and 1, leaving complex correlations unchanged
+"""
+clampcor(x::Real) = clamp(x, -1, 1)
+clampcor(x) = x
```

Comment:
With infinite precision, `abs(x) <= 1`, but if this isn't the case due to floating point rounding, I don't think there's a principled way to adjust `x`.